usaspending_api/etl/management/helpers/load_submission.py

```diff
@@ -0,0 +1,166 @@
+import numpy as np
+import pandas as pd
+
+from collections import deque
+from django.db import connections
+from django.utils.functional import cached_property
+from usaspending_api.references.models import ObjectClass, RefProgramActivity
+
+
+class Bunch:
+    """Generic class to hold a group of attributes."""
+
+    def __init__(self, **kwds):
+        self.__dict__.update(kwds)
+
+
+def get_or_create_program_activity(row, submission_attributes):
+    # We do it this way rather than .get_or_create because we do not want to duplicate existing pk's with null values
+    filters = {
+        "program_activity_code": row["program_activity_code"],
+        "program_activity_name": row["program_activity_name"].upper()
+        if row["program_activity_name"]
+        else row["program_activity_name"],
```

Comment:
I assume this is just to not try and uppercase an empty string? Just checking cause it looks like there might be a mistake here...

Reply:
Literally a straight copy.  None of these functions were touched.  In this file, the only things I'm responsible for are `CertifiedAwardFinancial` and `CertifiedAwardFinancialIterator`.

If you think it's a bug, let's make a note and loop back around after the dust settles... unless we're 100% sure it's a bug, it's super trivial to fix, and doesn't require any backfilling.

Reply:
Just as a side note, that code looks to me like it's trying to avoid uppercasing nulls (Nones).